samples/graphql/api/api.go

```diff
@@ -0,0 +1,37 @@
+package api
+
+import (
+	"encoding/json"
+	"net/http"
+	"time"
+)
+
+var appStartTime time.Time
+
+func init() {
+	appStartTime = time.Now()
+}
+
+func Status() func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+
+		type status struct {
+			Status string `json:"status"`
+			Redis  string `json:"redis"`
+			Uptime string `json:"uptime"`
+		}
+
+		s := status{
+			Status: func() string {
+				return "up"
+			}(),
+			Redis: func() string {
+				return "connected"
+			}(),
+			Uptime: time.Since(appStartTime).Truncate(time.Second).String(),
+		}
```

Comment:
Is this how Go translates an object to json?

Reply:
No, this is just defining the go struct. converting it to json and writing to stream is part of the `Encode` method two lines below (`json.NewEncoder(w).Encode(s)`)